test/integration/errors_test.go

```diff
@@ -183,3 +191,89 @@ func TestRejectedIdentifier(t *testing.T) {
 	test.AssertContains(t, prob.Detail, "Domain name contains an invalid character")
 	test.AssertContains(t, prob.Detail, "and 4 more problems")
 }
+
+// TestBadSignatureAlgorithm tests that supplying an unacceptable value for the
+// "alg" field of the JWS Protected Header results in a problem document with
+// the set of acceptable "alg" values listed in a custom extension field named
+// "algorithms". Creating a request with an unacceptable "alg" field requires
+// us to do some shenanigans.
+func TestBadSignatureAlgorithm(t *testing.T) {
+	t.Parallel()
+
+	client, err := makeClient()
+	if err != nil {
+		t.Fatal("creating test client")
+	}
+
+	header, err := json.Marshal(&struct {
+		Alg   string `json:"alg"`
+		KID   string `json:"kid"`
+		Nonce string `json:"nonce"`
+		URL   string `json:"url"`
+	}{
+		Alg:   string(jose.RS512), // This is the important bit; RS512 is unacceptable.
+		KID:   client.Account.URL,
+		Nonce: "deadbeef",
+		URL:   client.Directory().NewNonce,
+	})
+	if err != nil {
+		t.Fatalf("creating JWS protected header: %s", err)
+	}
+	phead := base64.RawStdEncoding.EncodeToString(header)
```

Comment:
Nit; per https://datatracker.ietf.org/doc/html/rfc7515#section-3.2 we should be using BASE64URL encoding here.